### PR TITLE
[NAE-1680] Public view + change behavior bug

### PR DESCRIPTION
**Pull request by renczesstefan**
# Description

There was missing code in public-task-resource.service.ts file, which is present in normal task-resource.service.ts. 

Fixes [NAE-1680]

## Dependencies

No new dependencies were introduced.

### Third party dependencies

No new dependencies were introduced.

### Blocking Pull requests

There are no dependencies on other PR.

## How Has Been This Tested?

This was tested manually.

### Test Configuration

| Name                | Tested on |
|---------------------| --------- |
| OS                  |   macOS Monterey 12.2.1        |
| Runtime             |  Node 14.19.3         |
| Dependency Manager  |  NPM 6.14.17        |
| Framework version   |  Angular 13.3.1         |
| Run parameters      |           |
| Other configuration |           |

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes have been checked, personally or remotely, with @mladoniczky 
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have resolved all conflicts with the target branch of the PR
- [x] I have updated and synced my code with the target branch
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes:
    - [x] Lint test
    - [x] Unit tests
    - [x] Integration tests
- [x] I have checked my contribution with code analysis tools:
    - [x] [SonarCloud](https://sonarcloud.io/project/overview?id=netgrif_application-engine)
    - [x] [Snyk](https://app.snyk.io/org/netgrif)
- [x] I have made corresponding changes to the documentation:
    - [x] Developer documentation
    - [x] User Guides
    - [x] Migration Guides

[NAE-1680]: https://netgrif.atlassian.net/browse/NAE-1680?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: projects/nae-example-app/src/app/doc/public-single-task-view/public-single-task-view.component.ts [hunk 1]
```diff
@@ -138,12 +144,20 @@ export class PublicSingleTaskViewComponent extends AbstractSingleTaskViewCompone
...
+        this._router.routeReuseStrategy.shouldReuseRoute = () => false
+    }
+
     ngAfterViewInit(): void {
         this.initializeHeader(this.taskHeaderComponent);
     }
 
     logEvent(event: TaskEventNotification) {
-        console.log(event);
+        if (!!event && event.event === TaskEvent.FINISH) {
```
Comment (minop) [inline projects/nae-example-app/src/app/doc/public-single-task-view/public-single-task-view.component.ts:156]: can be simplified to `event?.event === TaskEvent.FINISH`

File: projects/nae-example-app/src/app/doc/public-single-task-view/public-single-task-view.component.ts [hunk 2]
```diff
@@ -119,16 +124,17 @@ const caseResourceServiceFactory = (userService: UserService, sessionService: Se
         TaskRequestStateService,
         TaskEventService,
         {provide: NAE_TASK_OPERATIONS, useClass: SubjectTaskOperations},
+        {provide: AllowedNetsServiceFactory, useClass: AllowedNetsServiceFactory},
     ]
 })
-export class PublicSingleTaskViewComponent extends AbstractSingleTaskViewComponent implements AfterViewInit {
+export class PublicSingleTaskViewComponent extends AbstractSingleTaskViewComponent implements OnInit, AfterViewInit {
```
Comment (minop) [inline projects/nae-example-app/src/app/doc/public-single-task-view/public-single-task-view.component.ts:130]: changes were made to a component in the example-app. Does a schematic for generating these components exist? If yes, then its template files should be changed as well.

**Review by minop — DISMISSED** [review]

File: projects/nae-example-app/src/app/doc/public-single-task-view/public-single-task-view.component.ts (hunk 1 above)
Comment (renczesstefan) [reply projects/nae-example-app/src/app/doc/public-single-task-view/public-single-task-view.component.ts:156]: Sure, I will correct it. Thanks!

Comment (renczesstefan) [reply projects/nae-example-app/src/app/doc/public-single-task-view/public-single-task-view.component.ts:156]: I just checked, and this part is not even needed.

File: projects/nae-example-app/src/app/doc/public-single-task-view/public-single-task-view.component.ts (hunk 2 above)
Comment (renczesstefan) [reply projects/nae-example-app/src/app/doc/public-single-task-view/public-single-task-view.component.ts:129]: Schematics will be resolved in standalone tasks. Thanks for the reminder, I will resolve all management regarded tasks tomorrow (PR correction, task creation into Jira, etc).

**Review by minop — APPROVED** [review]

**Review by machacjozef — APPROVED** [review]

**Review by Kovy95 — APPROVED** [review]

**Review by mladoniczky — APPROVED** [review]

**Pull request merged**